### ISSUE TITLE
NameError: uninitialized constant ActiveRecord::ConnectionAdapters::AbstractAdapter

### ISSUE THREAD

**Issue by tagliala** [rgeo/activerecord-postgis-adapter#323]
happens while trying to run tests locally

I do not know if this is still necessary, since AR 6.1 is autoloading differently:

https://github.com/rails/rails/blob/6-1-stable/activerecord/lib/active_record/connection_adapters.rb#L8

May depend on rails/rails#37187

**Comment by tagliala** [comment]
I don't know what I'm doing, but with this change:

```diff
diff --git a/lib/active_record/connection_adapters/postgis_adapter.rb b/lib/active_record/connection_adapters/postgis_adapter.rb
index 5b9e339..c60ee07 100644
--- a/lib/active_record/connection_adapters/postgis_adapter.rb
+++ b/lib/active_record/connection_adapters/postgis_adapter.rb
@@ -7,13 +7,7 @@
 
 require "rgeo/active_record"
 
-# autoload AbstractAdapter to avoid circular require and void context warnings
-module ActiveRecord
-  module ConnectionAdapters
-    AbstractAdapter
-  end
-end
-
+require "active_record/connection_adapters"
 require "active_record/connection_adapters/postgresql_adapter"
 require "active_record/connection_adapters/postgis/version"
 require "active_record/connection_adapters/postgis/column_methods"
```

I get this results

```
Run options: --seed 65467

# Running:

.EEEEEEEEEEE...................................................

Fabulous run in 3.174601s, 19.8450 runs/s, 52.2900 assertions/s.

  1) Error:
TasksTest#test_add_index_via_references:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:118:in `test_add_index_via_references'

  2) Error:
TasksTest#test_add_index_with_no_options:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:107:in `test_add_index_with_no_options'

  3) Error:
TasksTest#test_basic_geography_schema_dump:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:79:in `test_basic_geography_schema_dump'

  4) Error:
TasksTest#test_basic_geometry_schema_dump:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:65:in `test_basic_geometry_schema_dump'

  5) Error:
TasksTest#test_create_database_from_extension_in_public_schema:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:153:in `rescue in drop_db_if_exists'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:151:in `drop_db_if_exists'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:7:in `test_create_database_from_extension_in_public_schema'

  6) Error:
TasksTest#test_create_database_from_extension_in_separate_schema:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:153:in `rescue in drop_db_if_exists'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:151:in `drop_db_if_exists'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:13:in `test_create_database_from_extension_in_separate_schema'

  7) Error:
TasksTest#test_empty_schema_dump:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:56:in `test_empty_schema_dump'

  8) Error:
TasksTest#test_empty_sql_dump:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:20:in `test_empty_sql_dump'

  9) Error:
TasksTest#test_index_schema_dump:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:93:in `test_index_schema_dump'

 10) Error:
TasksTest#test_index_sql_dump:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:41:in `test_index_sql_dump'

 11) Error:
TasksTest#test_sql_dump:
NameError: uninitialized constant ActiveRecord::Tasks::DatabaseAlreadyExists
Did you mean?  ActiveRecord::DatabaseAlreadyExists
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:147:in `rescue in setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:142:in `setup_database_tasks'
    ~/dev/activerecord-postgis-adapter/test/tasks_test.rb:27:in `test_sql_dump'

63 runs, 166 assertions, 0 failures, 11 errors, 0 skips
rake aborted!
Command failed with status (1)
~/.rvm/gems/ruby-2.7.2/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
~/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:24:in `eval'
~/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:24:in `<main>'
Tasks: TOP => default => test
(See full trace by running task with --trace)
```

**Comment by keithdoggett** [comment]
closed by #325
